modules/thermal_hydraulics/src/components/HeatTransferFromSpecifiedTemperature1Phase.C

```diff
@@ -33,7 +33,9 @@ HeatTransferFromSpecifiedTemperature1Phase::addVariables()
 {
   HeatTransferFromTemperature1Phase::addVariables();
 
-  getTHMProblem().addFunctionIC(_T_wall_name, _T_wall_fn_name, _flow_channel_subdomains);
+  if (!_app.isRestarting())
+    getTHMProblem().addFunctionIC(_T_wall_name, _T_wall_fn_name, _flow_channel_subdomains);
```

Comment:
same for all the non-zero ICs being born from user-input

Reply:
Same comment for T_wall